library/src/extra/rocsparse_csrgeam_nnz.cpp

```diff
@@ -916,6 +794,258 @@ namespace rocsparse
     }
 }
 
+rocsparse_status rocsparse::csrgeam_allocate_descr_memory(rocsparse_handle       handle,
+                                                          int64_t                m,
+                                                          int64_t                n,
+                                                          const void*            alpha,
+                                                          int64_t                nnz_A,
+                                                          const void*            beta,
+                                                          int64_t                nnz_B,
+                                                          rocsparse_spgeam_descr descr)
+{
+    ROCSPARSE_ROUTINE_TRACE;
+
+    // Clean up row pointer array
+    if(descr->csr_row_ptr_C != nullptr)
+    {
+        RETURN_IF_HIP_ERROR(rocsparse_hipFreeAsync(descr->csr_row_ptr_C, handle->stream));
+    }
+
+    // Clean up rocprim buffer
+    if(descr->rocprim_buffer != nullptr && descr->rocprim_alloc)
+    {
+        RETURN_IF_HIP_ERROR(rocsparse_hipFreeAsync(descr->rocprim_buffer, handle->stream));
+    }
+
+    descr->indextype = (nnz_A + nnz_B) < std::numeric_limits<int32_t>::max()
+                           ? rocsparse_indextype_i32
+                           : rocsparse_indextype_i64;
+
+    // We do not know how many nonzeros will exist in C yet therefore use an int64_t row ptr array
+    if(descr->indextype == rocsparse_indextype_i32)
+    {
+        RETURN_IF_HIP_ERROR(rocsparse_hipMallocAsync(
+            (void**)&descr->csr_row_ptr_C, sizeof(int32_t) * (m + 1), handle->stream));
```

Comment:
Can you use rocsparse_indextype_sizeof?

Reply:
(void**) is unnecessary in "rocsparse_hipMallocAsync( (void**)&descr->csr_row_ptr_C"